lib/flux/app-state.ts

```diff
@@ -145,11 +144,7 @@ export const actionMap = new ActionMap({
       });
     },
 
-    setEditorMode(state: AppState, { mode }: { mode: T.EditorMode }) {
-      return update(state, {
-        editorMode: { $set: mode },
-      });
-    },
+    setEditorMode() {},
```

Comment:
In line 225/220 below we do this…

```ts
if (settings.markdown) {
  dispatch(this.action('setEditorMode', { mode: 'edit' });
}
```

…which we can tell is superfluous and causes another dispatch cycle.

instead we can watch for `App.newNote` in our `editorMode` reducer and make sure that whenever we create a new note we reset to the edit mode.

```ts
const editorMode: A.Reducer<T.EditorMode> = (state = 'edit', action) => {
   switch (action.type) {
     case 'SET_EDITOR_MODE':
     case 'App.setEditorMode':
       return action.mode;
     case 'App.newNote':
       return 'edit'; 
     default:
       return state;
   }
 };
```

Reply:
<img width="189" alt="Screen Shot 2020-01-30 at 2 56 44 PM" src="https://user-images.githubusercontent.com/6817400/73485258-c775b680-4370-11ea-9c83-4948d7080477.png">

App.newNote does not fire on creating a new note.

Reply:
> For some reason I remember having an issue with that, maybe because newNote is a creator? 

That must be the problem I found. In that case do you have thoughts on creating a new proper action type (which at this point need not include any parameters or reducers) for `CREATE_NOTE` which we could then listen for?

We won't get rid of the double-dispatch but at least it would preserve the semantic of the triggering action - user created a note - which we could listen for, and which we can expand as we go to fill in (that new action).